### PR TITLE
[proposal] Improvements to the AWS CLI Contribution Process

### PR DESCRIPTION
**Pull request by kdaily**
The proposal specifies improvements to the community contribution process for the AWS CLI. For MarkDown rendered version of this proposal, view it [here](https://github.com/kdaily/aws-cli/blob/pr-contrib-proposal/proposals/pr-contrib-guide.md).

### ⚠️  Important note for community members

The AWS CLI development team is trying out a new process where proposals for new features are opened and reviewed as pull requests on GitHub. This allows us to be more open on our thought process and solicit design feedback from community members. Currently, we do not recommend community members from opening pull requests with new proposals until we have fully solidified and documented the proposal process. However, we do recommend and encourage community engagement by:

* Reading the proposal and providing feedback. If you are in general agreement with the proposal, please add the :+1: reaction to this top-level comment. If you have any questions or suggestions, feel free to leave a comment in the pull request.

### CODE REVIEW

File: proposals/pr-contrib-guide.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
+# Improvements to the AWS CLI Contribution Process
+
+Proposal         | Metadata
+---------------- | -------------
+**Author**       | Kenneth Daily
+**Status**       | Proposed
+**Created**      | 19-October-2021
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:7]: Let's update the created date to either today or when we plan to send a PR to the CLI repo.

File: proposals/pr-contrib-guide.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+Maintainers will regularly review open GitHub issues and label the ones that are
+ready for community contribution. The decision to label an issue as ready for
+community contribution is ultimately up to the discretion of the maintainers.
+Depending on the issue type (bug or feature request) and area (code or
+documentation), the maintainers will use a set of criteria to decide if the
+issue is ready for contribution. This may include:
+
+1. The change is feasible for a community member to contribute. See [the
+   appendix](#changes-not-available-for-contribution) for details.
+1. Feature requests must have at least 10 :+1: reactions. See the
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:187]: Did we want to change this from must to should?

File: proposals/pr-contrib-guide.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+from this queue. The issues will be prioritized for review publicly based on the
+following criteria in descending order of importance:
+
+1. Bugs, documentation changes, and feature requests will be prioritized in that
+   order. See the [rationale](#rationale-priority-order) for further discussion.
+1. Older contributions will be selected before more recent ones.
+
+A maintainer must select the issue from the queue with the current highest
+priority. The prioritization of issues in the queue is reviewed on a regular
+cadence and is ultimately decided based on the maintainers' discretion.
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:251]: We should add rationale link here for why we re-prioritize.

File: proposals/pr-contrib-guide.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+
+## Managing the existing backlog
+
+As of 2021-12-15, there are 188 pull requests and 436 issues in the AWS CLI
+GitHub repository. 43 pull requests and 87 issues have not been updated in more
+than a year. 95 issues have ten or more upvotes. Many of these issues are
+candidates for the ready to contribute or ready for review stages. We can use
+them to backfill the proposed contribution process, providing the team with pull
+requests to review and issues for the community to contribute.
+<!-- [TBD] open pull requests are directly related to an existing feature request. -->
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:304]: We should get rid of this comment.

File: proposals/pr-contrib-guide.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+
+We propose to manage existing pull requests in two phases. First, we will
+determine if there is a corresponding tracking issue for each pull request and
+[link them
+together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
+If no tracking issue exists, we will create one and link it to the pull request.
+Then, we will apply the new specification to all open issues identify the
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:311]: There probably needs to and `and` between `issues` and `identify`.

File: proposals/pr-contrib-guide.md [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+This project board will track GitHub issues from the contribution-ready stage
+through completion by merging a pull request. Each lane of the project board
+maps to a stage of the contribution process. New and existing GitHub labels and
+GitHub issue events will be used to drive the movement of the issue between the
+lanes of the GitHub project, which map to the stages of the contribution
+process. GitHub issue labels will be used to move an issue to the next lane.
+
+The contribution-ready lane is the entry point for users interested in making a
+contribution. The ready for review lane is the prioritization queue, and issues
+will be ordered in decreasing priority from top to bottom.
```
Comment (kyleknap) [inline proposals/pr-contrib-guide.md:293]: I still think any visualizations we can add here will be helpful whether it be a link to the board you have right now in the PR overview or a screenshot of it in the specification.

**Review by kyleknap — COMMENTED** [review]
Just had some small comments from my end to end pass of it. It would also be nice if we cleaned up the commit message to be a bit more formal.

File: proposals/pr-contrib-guide.md (hunk 2 above)
Comment (kdaily) [reply proposals/pr-contrib-guide.md:187]: yep.

**Pull request closed without merging**